core/src/com/unciv/ui/components/ParticleEffectAnimation.kt

```diff
@@ -0,0 +1,200 @@
+package com.unciv.ui.components
+
+import com.badlogic.gdx.Gdx
+import com.badlogic.gdx.graphics.GL20
+import com.badlogic.gdx.graphics.g2d.ParticleEffect
+import com.badlogic.gdx.graphics.g2d.ParticleEmitter
+import com.badlogic.gdx.graphics.g2d.ParticleEmitter.RangedNumericValue
+import com.badlogic.gdx.math.Interpolation
+import com.badlogic.gdx.math.Rectangle
+import com.badlogic.gdx.math.Vector2
+import com.badlogic.gdx.scenes.scene2d.Stage
+import com.badlogic.gdx.utils.Array
+import com.badlogic.gdx.utils.Disposable
+import com.unciv.UncivGame
+import com.unciv.ui.images.ImageGetter
+
+/** Hosts one template ParticleEffect and any number of clones, and each can travel linearly over part of its lifetime.
+ *  @property load Must be called from a subclass at least once - see detailed Kdoc
+ *  @property configure Optionally modifies all clones - see detailed Kdoc
+ *  @property getTargetBounds Where to draw on the stage - see detailed Kdoc
+ *  @property getScale Optionally scales all effects - see detailed Kdoc
+ *  @property render Needs to be called from the hosting Screen in its render override
+ *  @property dispose Required - this is just the Gdx Disposable interface, no automatic call
+ */
+abstract class ParticleEffectAnimation : Disposable {
```

Comment:
for files taken mostly from elsewhere, the accreditation should be on the files themselves